syncstorage/src/tokenserver/extractors.rs

```diff
@@ -98,6 +97,14 @@ impl TokenserverRequest {
             });
         }
 
+        // If the client previously reported a client state, every subsequent request must include
+        // one. Note that this is only relevant for BrowserID requests, since OAuth requests must
+        // always include a client state.
+        if !self.user.client_state.is_empty() && self.auth_data.client_state.is_empty() {
```

Comment:
This check needs to happen before the check for an already-used client state, since `""` could also be a previously used client state, which would cause `""` to invoke a "stale value" error instead of an "empty string" error. (The Python Tokenserver does this too: https://github.com/mozilla-services/tokenserver/blob/master/tokenserver/views.py#L419)

A small detail with likely no material impact, but best to keep the Rust Tokenserver as close as possible to the Python Tokenserver